src/copas.lua

```diff
@@ -368,6 +368,7 @@ function copas.dohandshake(skt, sslt)
   ssl = ssl or require("ssl")
   local nskt, err = ssl.wrap(skt, sslt)
   if not nskt then return error(err) end
+  if sslt.hostname then nskt:sni(sslt.hostname) end
```

Comment:
The call to `sni` has 2 signatures, 1 for clients and 1 for servers. This implements only the client version.

See https://github.com/brunoos/luasec/wiki/LuaSec-0.9#connsnimap--strict

I think this needs a test to see what type of socket we have, and then call the appropriate version of the `sni` function. I need to find some time to test how to do this exactly, I suspect something like this would give away whether it's a client or server: https://github.com/keplerproject/copas/blob/master/src/copas.lua#L200-L202

Reply:
hmmm... that doesn't give it away

Reply:
another thing to verify is that the call to `sni` results in a hard failure. An error may not go by unnoticed (same as with `wrap` just above), as that would be a security risk.

Reply:
At both https://github.com/brunoos/luasec/blob/master/samples/sni/client.lua and https://github.com/brunoos/luasec/blob/master/src/ssl.c#L754-L794 there are no hard failures, it always returns 0.